src/features/core/deviceModeTransformation/dmtHandler.js

```diff
@@ -0,0 +1,190 @@
+      const url = `${this.dataPlaneUrl}/transform`;
+      const headers = {
+        'Content-Type': 'application/json',
+        Authorization: `Basic ${btoa(`${this.writeKey}:`)}`,
+      };
+
+      try {
+        const xhr = new XMLHttpRequest();
+        xhr.open('POST', url, true);
+        Object.keys(headers).forEach((k) => xhr.setRequestHeader(k, headers[k]));
+
+        xhr.onreadystatechange = () => {
+          if (xhr.readyState === 4) {
+            try {
+              const { status } = xhr;
+              let { response } = xhr;
+
+              if (status === 200) {
+                if (response && typeof response === 'string') {
+                  response = JSON.parse(response);
+                } else {
+                  reject(`[Transformation]:: Transformation failed. Invalid response from server.`);
+                  return;
+                }
+                /**
+                 * Sample Response format:
+                 * {
+                    "transformedBatch" :[
+                      {
+                        "id": "destination-id",
+                        "payload": [
+                          {
+                            "orderNo":1,
+                            "status": "200",
+                            "event": {
+                              "message": { ...}
+                          }]
+                      }]
+                    } 
+                */
+                /**
+                 * Filter the successful transformed event
+                 */
+                const transformationResponse = [];
+                response.transformedBatch.forEach((dest) => {
+                  transformationResponse.push({
+                    id: dest.id,
+                    payload: dest.payload.filter((tEvent) => tEvent.status === '200'),
+                  });
+                });
+
+                resolve({
+                  transformedPayload: transformationResponse,
+                  transformationServerAccess: true,
+                });
+                return;
+              }
+              if (status === 400) {
+                const errorMessage = response
+                  ? `[Transformation]:: ${response}`
+                  : `[Transformation]:: Invalid request payload`;
+                reject(errorMessage);
+                return;
+              }
+              if (status === 404) {
+                resolve({
+                  transformedPayload: payload.batch,
+                  transformationServerAccess: false,
+                });
+                return;
+              }
+
+              // If the request is not successful
+              // one or more transformation is unsuccessfull
+              // retry till the retryAttempt is exhausted
+              if (retryAttempt > 0) {
+                const newretryAttempt = retryAttempt - 1;
+                setTimeout(() => this.sendEventForTransformation(payload, newretryAttempt)
```

Comment:
should we retry with a backoff like our event queue or is this how the mobile sdks did it too?

Reply:
Adressed